packages/svelte/src/internal/client/runtime.js

```diff
@@ -361,16 +357,10 @@ function remove_reaction(signal, dependency) {
 	const reactions = dependency.reactions;
 	let reactions_length = 0;
 	if (reactions !== null) {
-		reactions_length = reactions.length - 1;
-		const index = reactions.indexOf(signal);
-		if (index !== -1) {
-			if (reactions_length === 0) {
-				dependency.reactions = null;
-			} else {
-				// Swap with last element and then remove.
-				reactions[index] = reactions[reactions_length];
-				reactions.pop();
-			}
+		reactions.delete(signal);
+		reactions_length = reactions.size;
+		if (reactions_length === 0) {
```

Comment:
i think we can get rid of the `reactions_length` variable now, and instead of checking `reactions_length === 0` on line 368, just do `reactions === null`?

```suggestion
		if (reactions.size === 0) {
```